src/patterns/shaders/lissajous.frag

```diff
@@ -0,0 +1,43 @@
+#include <conjurer_common>
+
+#ifdef GL_ES
+precision mediump float;
+#endif
+
+varying vec2 v_uv;
+uniform float u_time;
+uniform vec2 u_resolution;
+
+uniform float u_count;
+uniform float u_space;
+uniform float u_a0;
+uniform float u_a1;
+uniform float u_b0;
+uniform float u_b1;
+uniform Palette u_palette;
+
+void main() {
+    vec2 uv = v_uv;
+
+    // Convert from canopy space to cartesian
+    float theta = uv.x * 2.0 * 3.1415926;
+    float r = uv.y * 0.88888888 + 0.111111111;
+    float x = r * cos(theta) * 0.5 + 0.5;
+    float y = r * sin(theta) * 0.5 + 0.5;
+    uv = vec2(x, y) * 2.0 - 1.0;
+
+    vec3 col = vec3(0.);
+    float c = u_count < 1. ? 1. : u_count; // todo: int props? editable min/max values on props
```

Comment:
yeah I'm thinking of specifying in the definition of the uniform something along the lines of:

```
    u_count: {
      name: "Count",
      value: 8,
      min: 1,
      max: 16,
      step: 1
    },
```

Then when you are playing with it in the playground, we can use those values when setting up the sliders